packages/hardhat-core/src/internal/hardhat-network/provider/RethnetState.ts

```diff
@@ -29,14 +30,17 @@ export class RethnetStateManager {
     );
   }
 
-  public static forkRemote(
+  public static async forkRemote(
     context: RethnetContext,
     forkConfig: ForkConfig,
     genesisAccounts: GenesisAccount[]
-  ): RethnetStateManager {
+  ): Promise<RethnetStateManager> {
     let blockNumber;
     if (forkConfig.blockNumber !== undefined) {
       blockNumber = BigInt(forkConfig.blockNumber);
+    } else {
+      const { forkBlockNumber } = await makeForkProvider(forkConfig);
+      blockNumber = forkBlockNumber;
```

Comment:
Just out of curiosity: why get the latest fork block number out here instead of within the rust side of things?

Reply:
The logic that's used to determine the block number is very specific and already exists on the EthereumJS side. It was safer to just re-use that instead of trying to reproduce the code to do so.

The previous code was causing issues with block number mismatches